server/src/main/proto/spine/system/server/c/command_lifecycle_events.proto

```diff
@@ -0,0 +1,97 @@
+//
+// Copyright 2018, TeamDev. All rights reserved.
+//
+// Redistribution and use in source and/or binary forms, with or without
+// modification, must retain the above copyright notice and the following
+// disclaimer.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+syntax = "proto3";
+
+package spine.system.server;
+
+import "spine/options.proto";
+
+option (type_url_prefix) = "type.spine.io";
+option (internal_all) = true;
+
+option java_package = "io.spine.system.server";
+option java_outer_classname = "CommandLifecycleEventsProto";
+option java_multiple_files = true;
+
+import "spine/base/error.proto";
+import "spine/core/command.proto";
+import "spine/core/rejection.proto";
+import "spine/client/entities.proto";
+
+// An event emmited when a new command is received.
+//
+message CommandReceived {
+
+    // The ID of the command.
+    spine.core.CommandId id = 1;
+
+    // The command.
+    spine.core.Command payload = 2;
+}
+
+// An event emmited when the command is acknowledged by the `CommandBus`.
+//
+message CommandAcknowledged {
+
+    // The ID of the command.
+    spine.core.CommandId id = 1;
+}
+
+// An event emmited when a command is dispatched to a handler
+//
+message CommandDispatched {
+
+    // The ID of the command.
+    spine.core.CommandId id = 1;
+
+    // The ID of the entity which the command is dispatched to.
+    spine.client.EntityId receiver = 2;
```

Comment:
I think, it's `target`.

Reply:
This field will be deleted in part 2. Let's just leave it as is right now.